src/librustc_back/target/mod.rs

```diff
@@ -168,6 +168,8 @@ pub struct TargetOptions {
     /// currently only "gnu" is used to fall into LLVM. Unknown strings cause
     /// the system linker to be used.
     pub archive_format: String,
+    /// Is asm!() not allowed? Defaults to false
+    pub no_asm: bool,
```

Comment:
Could this instead be inverted to `allow_asm`? Reading a double negative is sometimes a little confusing. Other than that though looks good to me!